### PR TITLE
Bug 1074117 - Use expected conditions where appropriate and optimise ele...

### PR DESCRIPTION
**Pull request by mwargers**
...ment lookups, fix system app

### CODE REVIEW

File: tests/python/gaia-ui-tests/gaiatest/apps/system/app.py [hunk 1]
```diff
@@ -23,25 +26,38 @@ class System(Base):
     _software_home_button_locator = (By.ID, 'software-home-button')
 
     def wait_for_status_bar_displayed(self):
-        self.wait_for_element_displayed(*self._status_bar_locator)
+        element = Wait(self.marionette).until(
+            expected.element_present(*self._status_bar_locator))
+        Wait(self.marionette).until(expected.element_displayed(element))
```
Comment (davehunt) [inline tests/python/gaia-ui-tests/gaiatest/apps/system/app.py]: `element_displayed` can now take a locator, so this can be written as:

```python
Wait(self.marionette).until(expected.element_displayed(*self._status_bar_locator)
```

File: tests/python/gaia-ui-tests/gaiatest/apps/system/app.py [hunk 2]
```diff
@@ -23,25 +26,38 @@ class System(Base):
...
 
     def wait_for_notification_toaster_displayed(self, timeout=10, message=None):
-        self.wait_for_element_displayed(*self._notification_toaster_locator)
+        element = Wait(self.marionette).until(
+            expected.element_present(*self._notification_toaster_locator))
+        Wait(self.marionette, timeout=timeout).until(
+            expected.element_displayed(element), message=message)
```
Comment (davehunt) [inline tests/python/gaia-ui-tests/gaiatest/apps/system/app.py]: As above

File: tests/python/gaia-ui-tests/gaiatest/apps/system/app.py [hunk 3]
```diff
@@ -23,25 +26,38 @@ class System(Base):
...
 
     def wait_for_notification_toaster_not_displayed(self, timeout=10):
-        self.wait_for_element_not_displayed(*self._notification_toaster_locator)
+        Wait(self.marionette, timeout=timeout, ignored_exceptions=StaleElementException).until(
+            lambda m: not m.find_element(*self._notification_toaster_locator).is_displayed())
```
Comment (davehunt) [inline tests/python/gaia-ui-tests/gaiatest/apps/system/app.py]: These aren't great methods because they don't give any confidence that the element in question ever was displayed (or is not yet to be displayed). That said, let's not concern ourselves with fixing this here. You can simply use expected.element_not_displayed, which takes hidden, missing and stale elements into account:

```python
Wait(self.marionette, timeout).until(
    expected.element_not_displayed(*self._notification_toaster_locator)
```

File: tests/python/gaia-ui-tests/gaiatest/apps/system/app.py [hunk 4]
```diff
@@ -23,25 +26,38 @@ class System(Base):
...
 
     def wait_for_system_banner_displayed(self):
-        self.wait_for_element_displayed(*self._system_banner_locator)
+        element = Wait(self.marionette).until(
+            expected.element_present(*self._system_banner_locator))
+        Wait(self.marionette).until(expected.element_displayed(element))
```
Comment (davehunt) [inline tests/python/gaia-ui-tests/gaiatest/apps/system/app.py]: As above, we can condense this now.

File: tests/python/gaia-ui-tests/gaiatest/apps/system/app.py [hunk 5]
```diff
@@ -23,25 +26,38 @@ class System(Base):
...
-        self.wait_for_element_not_displayed(*self._system_banner_locator)
+        Wait(self.marionette).until(
+            expected.element_not_displayed(*self._system_banner_locator))
 
     def wait_for_software_home_button_displayed(self, timeout=10, message=None):
-        self.wait_for_element_displayed(*self._software_home_button_locator)
+        element = Wait(self.marionette).until(
+            expected.element_present(*self._software_home_button_locator))
+        Wait(self.marionette, timeout=timeout).until(
+            expected.element_displayed(element), message=message)
```
Comment (davehunt) [inline tests/python/gaia-ui-tests/gaiatest/apps/system/app.py]: As above, this can be condensed.

File: tests/python/gaia-ui-tests/gaiatest/apps/system/app.py [hunk 6]
```diff
@@ -61,11 +77,15 @@ def is_app_update_notification_displayed(self):
         return update_manager_toaster.location['y'] > (0 - update_manager_toaster.size['height'])
 
     def wait_for_app_update_to_clear(self):
-        update_manager_toaster = self.marionette.find_element(*self._update_manager_toaster_locator)
-        self.wait_for_condition(lambda m: update_manager_toaster.location['y'] == (0 - update_manager_toaster.size['height']))
+        element = self.marionette.find_element(*self._update_manager_toaster_locator)
+        Wait(self.marionette).until(lambda m: element.location['y'] == (0 - update_manager_toaster.size['height']))
```
Comment (davehunt) [inline tests/python/gaia-ui-tests/gaiatest/apps/system/app.py]: Reference to `update_manager_toaster` here should be `element`?

File: tests/python/gaia-ui-tests/gaiatest/apps/system/app.py [hunk 7]
```diff
@@ -61,11 +77,15 @@ def is_app_update_notification_displayed(self):
...
 
     def wait_for_geolocation_icon_displayed(self):
-        self.wait_for_element_displayed(*self._geoloc_statusbar_locator, timeout=40000)
+        element = Wait(self.marionette).until(
+            expected.element_present(*self._geoloc_statusbar_locator))
+        Wait(self.marionette, timeout=40000).until(expected.element_displayed(element))
```
Comment (davehunt) [inline tests/python/gaia-ui-tests/gaiatest/apps/system/app.py]: We can condense this.

File: tests/python/gaia-ui-tests/gaiatest/apps/system/app.py [hunk 8]
```diff
@@ -61,11 +77,15 @@ def is_app_update_notification_displayed(self):
...
 
     def wait_for_airplane_mode_icon_displayed(self):
-        self.wait_for_element_displayed(*self._airplane_mode_statusbar_locator)
+        element = Wait(self.marionette).until(
+            expected.element_present(*self._airplane_mode_statusbar_locator))
+        Wait(self.marionette).until(expected.element_displayed(element))
```
Comment (davehunt) [inline tests/python/gaia-ui-tests/gaiatest/apps/system/app.py]: As above.

File: tests/python/gaia-ui-tests/gaiatest/apps/system/regions/activities.py [hunk 9]
```diff
@@ -30,35 +30,39 @@ def __init__(self, marionette):
 
     def tap_wallpaper(self):
         self.marionette.find_element(*self._wallpaper_button_locator).tap()
-        self.wait_for_element_not_displayed(*self._actions_menu_locator)
+        Wait(self.marionette).until(
+            expected.element_not_displayed(*self._actions_menu_locator))
```
Comment (davehunt) [inline tests/python/gaia-ui-tests/gaiatest/apps/system/regions/activities.py]: If this element is visible before we tap the wallpaper button then I would recommend finding it before the tap. This will allow the test to fail if the element locator changes.

File: tests/python/gaia-ui-tests/gaiatest/apps/system/regions/activities.py [hunk 10]
```diff
@@ -30,35 +30,39 @@ def __init__(self, marionette):
...
-        self.wait_for_condition(lambda m: self.apps.displayed_app.name == wallpaper.name)
+        Wait(self.marionette).until(lambda m: self.apps.displayed_app.name == wallpaper.name)
         self.apps.switch_to_displayed_app()
         return wallpaper
 
     def tap_gallery(self):
         self.marionette.find_element(*self._gallery_button_locator).tap()
-        self.wait_for_element_not_displayed(*self._actions_menu_locator)
+        Wait(self.marionette).until(
+            expected.element_not_displayed(*self._actions_menu_locator))
```
Comment (davehunt) [inline tests/python/gaia-ui-tests/gaiatest/apps/system/regions/activities.py]: As above, find the element before the tap (ideally verify that it's displayed using a wait), and then wait for it to be not displayed after.

File: tests/python/gaia-ui-tests/gaiatest/apps/system/regions/activities.py [hunk 11]
```diff
@@ -30,35 +30,39 @@ def __init__(self, marionette):
...
-        self.wait_for_condition(lambda m: self.apps.displayed_app.name == gallery.name)
+        Wait(self.marionette).until(lambda m: self.apps.displayed_app.name == gallery.name)
         self.apps.switch_to_displayed_app()
         return gallery
 
     def tap_camera(self):
         self.marionette.find_element(*self._camera_button_locator).tap()
-        self.wait_for_element_not_displayed(*self._actions_menu_locator)
+        Wait(self.marionette).until(
+            expected.element_not_displayed(*self._actions_menu_locator))
```
Comment (davehunt) [inline tests/python/gaia-ui-tests/gaiatest/apps/system/regions/activities.py]: As above.

File: tests/python/gaia-ui-tests/gaiatest/apps/system/regions/activities.py [hunk 12]
```diff
@@ -30,35 +30,39 @@ def __init__(self, marionette):
...
+        Wait(self.marionette).until(lambda m: self.apps.displayed_app.name == camera.name)
         self.apps.switch_to_displayed_app()
         camera.wait_for_capture_ready()
         return camera
 
     def tap_cancel(self):
         self.marionette.find_element(*self._cancel_button_locator).tap()
-        self.wait_for_element_not_displayed(*self._actions_menu_locator)
+        Wait(self.marionette).until(
+            expected.element_not_displayed(*self._actions_menu_locator))
```
Comment (davehunt) [inline tests/python/gaia-ui-tests/gaiatest/apps/system/regions/activities.py]: As above.

File: tests/python/gaia-ui-tests/gaiatest/apps/system/regions/activities.py [hunk 13]
```diff
@@ -76,6 +80,7 @@ def is_menu_visible(self):
 
     def share_to_messages(self):
         self.marionette.find_element(*self._messages_button_locator).tap()
-        self.wait_for_element_not_displayed(*self._actions_menu_locator)
+        Wait(self.marionette).until(
+            expected.element_not_displayed(*self._actions_menu_locator))
```
Comment (davehunt) [inline tests/python/gaia-ui-tests/gaiatest/apps/system/regions/activities.py]: As above.

File: tests/python/gaia-ui-tests/gaiatest/apps/system/regions/cards_view.py [hunk 14]
```diff
@@ -65,19 +67,22 @@ def is_app_present(self, app):
...
-        self.wait_for_condition(lambda m: self.is_app_displayed(app))
+        Wait(self.marionette).until(lambda m: self.is_app_displayed(app))
         self.marionette.find_element(*self._app_card_locator(app)).tap()
 
     def close_app(self, app):
         self.wait_for_card_ready(app)
         self.marionette.find_element(*self._close_button_locator(app)).tap()
-        self.wait_for_element_not_present(*self._app_card_locator(app))
+        Wait(self.marionette).until(
+            expected.element_not_present(*self._app_card_locator(app)))
```
Comment (davehunt) [inline tests/python/gaia-ui-tests/gaiatest/apps/system/regions/cards_view.py]: Locate the element before we close it, that way we verify it was visible and can wait for it to be not visible.

File: tests/python/gaia-ui-tests/gaiatest/apps/system/regions/cards_view.py [hunk 15]
```diff
@@ -65,19 +67,22 @@ def is_app_present(self, app):
...
 
     def wait_for_cards_view(self):
-        self.wait_for_condition(lambda m: m.find_element(*self._cards_view_locator).get_attribute('class') == 'active')
+        Wait(self.marionette).until(
+            lambda m: m.find_element(*self._cards_view_locator).get_attribute('class') == 'active')
```
Comment (davehunt) [inline tests/python/gaia-ui-tests/gaiatest/apps/system/regions/cards_view.py]: Find the element outside of the Wait so we do not repeatedly look it up.

File: tests/python/gaia-ui-tests/gaiatest/apps/system/regions/iac_publisher.py [hunk 16]
```diff
@@ -18,7 +20,9 @@ class IacPublisher(Base):
 
     def launch(self):
         Base.launch(self, launch_timeout=120000)
-        self.wait_for_element_displayed(*self._pub_app_msg_to_send_locator)
+        element = Wait(self.marionette).until(
+            expected.element_present(*self._pub_app_msg_to_send_locator))
+        Wait(self.marionette).until(expected.element_displayed(element))
```
Comment (davehunt) [inline tests/python/gaia-ui-tests/gaiatest/apps/system/regions/iac_publisher.py]: We can just use `element_displayed` with the locator to avoid the two waits.

File: tests/python/gaia-ui-tests/gaiatest/apps/system/regions/iac_publisher.py [hunk 17]
```diff
@@ -30,8 +34,10 @@ def tap_send_message(self):
         self.marionette.find_element(*self._pub_app_send_button_locator).tap()
 
     def wait_for_message_received(self):
-        self.wait_for_element_present(*self._pub_app_received_str_msg_locator) and \
-        self.wait_for_element_present(*self._pub_app_received_blob_msg_locator)
+        Wait(self.marionette).until(
+            expected.element_present(*self._pub_app_received_str_msg_locator)) and \
+        Wait(self.marionette).until(
+            expected.element_present(*self._pub_app_received_blob_msg_locator))
```
Comment (davehunt) [inline tests/python/gaia-ui-tests/gaiatest/apps/system/regions/iac_publisher.py]: I don't think the `and \` is doing anything here.

File: tests/python/gaia-ui-tests/gaiatest/apps/system/regions/sleep_view.py [hunk 18]
```diff
@@ -19,7 +21,9 @@ def is_menu_visible(self):
         return self.is_element_displayed(*self._sleep_menu_locator)
 
     def wait_for_sleep_menu_visible(self):
-        self.wait_for_element_displayed(*self._sleep_menu_locator)
+        element = Wait(self.marionette).until(
+            expected.element_present(*self._sleep_menu_locator))
+        Wait(self.marionette).until(expected.element_displayed(element))
```
Comment (davehunt) [inline tests/python/gaia-ui-tests/gaiatest/apps/system/regions/sleep_view.py]: We can condense this.

File: tests/python/gaia-ui-tests/gaiatest/apps/system/regions/time_picker.py [hunk 19]
```diff
@@ -23,15 +25,18 @@ class TimePicker(Base):
     def __init__(self, marionette):
         Base.__init__(self, marionette)
         self.marionette.switch_to_frame()
-        self.wait_for_element_displayed(*self._current_element(*self._hour_picker_locator))
+        element = Wait(self.marionette).until(
+            expected.element_present(*self._current_element(*self._hour_picker_locator)))
+        Wait(self.marionette).until(expected.element_displayed(element))
```
Comment (davehunt) [inline tests/python/gaia-ui-tests/gaiatest/apps/system/regions/time_picker.py]: We can condense this.

File: tests/python/gaia-ui-tests/gaiatest/apps/system/regions/time_picker.py [hunk 20]
```diff
@@ -23,15 +25,18 @@ class TimePicker(Base):
...
-        # TODO: wait for the time piker to fade in Bug 1038186
+        # TODO: wait for the time picker to fade in Bug 1038186
         time.sleep(2)
 
     def tap_done(self):
         self.marionette.find_element(*self._done_button_locator).tap()
-        self.wait_for_element_not_displayed(*self._time_picker_locator)
-        # TODO: wait for the time piker to fade out Bug 1038186
+        Wait(self.marionette).until(
+            expected.element_not_displayed(*self._time_picker_locator))
```
Comment (davehunt) [inline tests/python/gaia-ui-tests/gaiatest/apps/system/regions/time_picker.py]: If this is present/visible before the tap then find it before and wait for it to be not displayed after.

File: tests/python/gaia-ui-tests/gaiatest/apps/system/app.py (hunk 1 above)
Comment (mwargers) [inline tests/python/gaia-ui-tests/gaiatest/apps/system/app.py]: Heh, ok, I need to change this in quite a few places then.

File: tests/python/gaia-ui-tests/gaiatest/apps/system/app.py (hunk 3 above)
Comment (mwargers) [inline tests/python/gaia-ui-tests/gaiatest/apps/system/app.py]: I had to use the ignored_exceptions=StaleElementException thing, because it would otherwise time out on that spot.
I'll try again, once I addressed all your review comments.

File: tests/python/gaia-ui-tests/gaiatest/apps/system/regions/iac_publisher.py (hunk 17 above)
Comment (mwargers) [inline tests/python/gaia-ui-tests/gaiatest/apps/system/regions/iac_publisher.py]: Yeah, me neither, I'll remove it then.

File: tests/python/gaia-ui-tests/gaiatest/apps/system/app.py (hunk 6 above)
Comment (mwargers) [inline tests/python/gaia-ui-tests/gaiatest/apps/system/app.py]: Oops! Thanks!

File: tests/python/gaia-ui-tests/gaiatest/apps/system/regions/time_picker.py (hunk 20 above)
Comment (mwargers) [inline tests/python/gaia-ui-tests/gaiatest/apps/system/regions/time_picker.py]: Do you want a check before the tap() call for _time_picker_locator to be visible?

**Comment by try-server-hook** [comment]
[mwargers Martijn (mwargers)](https://github.com/mwargers) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=4133f3fa4e33)

File: tests/python/gaia-ui-tests/gaiatest/apps/system/regions/cards_view.py [hunk 21]
```diff
@@ -3,6 +3,8 @@
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
 import time
+from marionette import expected
```
Comment (mwargers) [inline tests/python/gaia-ui-tests/gaiatest/apps/system/regions/cards_view.py]: Ah, ok, I didn't know that was the convention.

Comment (davehunt) [inline tests/python/gaia-ui-tests/gaiatest/apps/system/regions/cards_view.py]: See http://legacy.python.org/dev/peps/pep-0008/#imports

File: tests/python/gaia-ui-tests/gaiatest/apps/system/app.py [hunk 22]
```diff
@@ -23,25 +25,30 @@ class System(Base):
     _software_home_button_locator = (By.ID, 'software-home-button')
 
     def wait_for_status_bar_displayed(self):
-        self.wait_for_element_displayed(*self._status_bar_locator)
+        Wait(self.marionette).until(expected.element_displayed(*self._status_bar_locator))
 
     def wait_for_notification_toaster_displayed(self, timeout=10, message=None):
-        self.wait_for_element_displayed(*self._notification_toaster_locator)
+        Wait(self.marionette, timeout=timeout).until(
+            expected.element_displayed(*self._notification_toaster_locator), message=message)
```
Comment (mwargers) [inline tests/python/gaia-ui-tests/gaiatest/apps/system/app.py]: Thanks, I didn't know (I don't know Python well, actually).

**Comment by try-server-hook** [comment]
[mwargers Martijn (mwargers)](https://github.com/mwargers) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=5d2832973025)

**Comment by try-server-hook** [comment]
[mwargers Martijn (mwargers)](https://github.com/mwargers) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=f709ed59d0d3)

File: tests/python/gaia-ui-tests/gaiatest/apps/system/app.py [hunk 23]
```diff
@@ -23,25 +25,30 @@ class System(Base):
     _software_home_button_locator = (By.ID, 'software-home-button')
 
     def wait_for_status_bar_displayed(self):
-        self.wait_for_element_displayed(*self._status_bar_locator)
+        Wait(self.marionette).until(expected.element_displayed(*self._status_bar_locator))
 
     def wait_for_notification_toaster_displayed(self, timeout=10, message=None):
-        self.wait_for_element_displayed(*self._notification_toaster_locator)
+        Wait(self.marionette, timeout).until(
+            expected.element_displayed(*self._notification_toaster_locator), message=message)
```
Comment (davehunt) [inline tests/python/gaia-ui-tests/gaiatest/apps/system/app.py]: You still have `message=message` where just `message` would do, but it's just a nit, and there's no need to change it.

File: tests/python/gaia-ui-tests/gaiatest/apps/system/regions/activities.py [hunk 24]
```diff
@@ -29,36 +29,48 @@ def __init__(self, marionette):
...
+        Wait(self.marionette).until(lambda m: self.apps.displayed_app.name == camera.name)
         self.apps.switch_to_displayed_app()
         camera.wait_for_capture_ready()
         return camera
 
     def tap_cancel(self):
         self.marionette.find_element(*self._cancel_button_locator).tap()
-        self.wait_for_element_not_displayed(*self._actions_menu_locator)
+        actions_menu = Wait(self.marionette).until(
+            expected.element_present(*self._actions_menu_locator))
```
Comment (davehunt) [inline tests/python/gaia-ui-tests/gaiatest/apps/system/regions/activities.py]: This is still after the tap, it should be before.

File: tests/python/gaia-ui-tests/gaiatest/apps/system/regions/activities.py [hunk 25]
```diff
@@ -76,6 +88,9 @@ def is_menu_visible(self):
 
     def share_to_messages(self):
         self.marionette.find_element(*self._messages_button_locator).tap()
-        self.wait_for_element_not_displayed(*self._actions_menu_locator)
+        actions_menu = Wait(self.marionette).until(
+            expected.element_present(*self._actions_menu_locator))
```
Comment (davehunt) [inline tests/python/gaia-ui-tests/gaiatest/apps/system/regions/activities.py]: Again, this should be before the tap.

File: tests/python/gaia-ui-tests/gaiatest/apps/system/regions/utility_tray.py [hunk 26]
```diff
@@ -41,8 +41,8 @@ def tap_settings_button(self):
         self.marionette.find_element(*self._quicksettings_app_locator).tap()
 
     def a11y_wheel_utility_tray_grippy(self):
-        self.accessibility.wheel(self.marionette.find_element(
-            *self._grippy_locator), 'up')
+        Wait(self.marionette).until(
+            expected.element_not_displayed(*System(self.marionette)._utility_tray_locator))
         self.wait_for_element_not_displayed(*System(self.marionette)._utility_tray_locator)
```
Comment (davehunt) [inline tests/python/gaia-ui-tests/gaiatest/apps/system/regions/utility_tray.py]: You need to remove this line.

File: tests/python/gaia-ui-tests/gaiatest/apps/system/regions/utility_tray.py [hunk 27]
```diff
@@ -41,8 +41,8 @@ def tap_settings_button(self):
         self.marionette.find_element(*self._quicksettings_app_locator).tap()
 
     def a11y_wheel_utility_tray_grippy(self):
-        self.accessibility.wheel(self.marionette.find_element(
-            *self._grippy_locator), 'up')
```
Comment (davehunt) [inline tests/python/gaia-ui-tests/gaiatest/apps/system/regions/utility_tray.py]: This removal doesn't appear to be related.

File: tests/python/gaia-ui-tests/gaiatest/apps/system/regions/activities.py (hunk 24 above)
Comment (mwargers) [inline tests/python/gaia-ui-tests/gaiatest/apps/system/regions/activities.py]: Sorry, didn't look at other places where I had to change this.

File: tests/python/gaia-ui-tests/gaiatest/apps/system/regions/utility_tray.py (hunk 26 above)
Comment (mwargers) [inline tests/python/gaia-ui-tests/gaiatest/apps/system/regions/utility_tray.py]: Oops!

File: tests/python/gaia-ui-tests/gaiatest/apps/system/regions/utility_tray.py (hunk 27 above)
Comment (mwargers) [inline tests/python/gaia-ui-tests/gaiatest/apps/system/regions/utility_tray.py]: Yes, I removed the wrong line(I managed to mix this up with the line that I didn't remove)

File: tests/python/gaia-ui-tests/gaiatest/apps/system/regions/activities.py [hunk 28]
```diff
@@ -29,36 +29,48 @@ def __init__(self, marionette):
...
         self.apps.switch_to_displayed_app()
         return wallpaper
 
     def tap_gallery(self):
         self.marionette.find_element(*self._gallery_button_locator).tap()
-        self.wait_for_element_not_displayed(*self._actions_menu_locator)
+        actions_menu = Wait(self.marionette).until(
+            expected.element_present(*self._actions_menu_locator))
+        Wait(self.marionette).until(
+            expected.element_not_displayed(actions_menu))
```
Comment (davehunt) [inline tests/python/gaia-ui-tests/gaiatest/apps/system/regions/activities.py]: This still doesn't find the element or wait for it to be visible before we tap it. If the element is never visible then this will still pass, even though our expectation is for it to transition from visible to not hidden.

File: tests/python/gaia-ui-tests/gaiatest/apps/system/regions/activities.py [hunk 29]
```diff
@@ -29,36 +29,48 @@ def __init__(self, marionette):
...
         self.apps.switch_to_displayed_app()
         return gallery
 
     def tap_camera(self):
         self.marionette.find_element(*self._camera_button_locator).tap()
-        self.wait_for_element_not_displayed(*self._actions_menu_locator)
+        actions_menu = Wait(self.marionette).until(
+            expected.element_present(*self._actions_menu_locator))
+        Wait(self.marionette).until(
+            expected.element_not_displayed(actions_menu))
```
Comment (davehunt) [inline tests/python/gaia-ui-tests/gaiatest/apps/system/regions/activities.py]: As above.

File: tests/python/gaia-ui-tests/gaiatest/apps/system/regions/activities.py [hunk 30]
```diff
@@ -29,36 +29,48 @@ def __init__(self, marionette):
...
         camera = Camera(self.marionette)
-        self.wait_for_condition(lambda m: self.apps.displayed_app.name == camera.name)
+        Wait(self.marionette).until(lambda m: self.apps.displayed_app.name == camera.name)
         self.apps.switch_to_displayed_app()
         camera.wait_for_capture_ready()
         return camera
 
     def tap_cancel(self):
+        actions_menu = Wait(self.marionette).until(
+            expected.element_present(*self._actions_menu_locator))
```
Comment (davehunt) [inline tests/python/gaia-ui-tests/gaiatest/apps/system/regions/activities.py]: I'd prefer if we waited for the element to be visible, as we're explicitly waiting based on visibility after the tap.

File: tests/python/gaia-ui-tests/gaiatest/apps/system/regions/activities.py [hunk 31]
```diff
@@ -75,7 +87,10 @@ def is_menu_visible(self):
         return self.is_element_displayed(*self._actions_menu_locator)
 
     def share_to_messages(self):
+        actions_menu = Wait(self.marionette).until(
+            expected.element_present(*self._actions_menu_locator))
```
Comment (davehunt) [inline tests/python/gaia-ui-tests/gaiatest/apps/system/regions/activities.py]: As above regarding visibility.

**Comment by try-server-hook** [comment]
[mwargers Martijn (mwargers)](https://github.com/mwargers) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=6e847c8624bb)

File: tests/python/gaia-ui-tests/gaiatest/apps/system/regions/cards_view.py [hunk 32]
```diff
@@ -65,19 +67,25 @@ def is_app_present(self, app):
...
         self.marionette.find_element(*self._app_card_locator(app)).tap()
 
     def close_app(self, app):
         self.wait_for_card_ready(app)
+        app_card = Wait(self.marionette).until(
+            expected.element_present(*self._app_card_locator(app)))
         self.marionette.find_element(*self._close_button_locator(app)).tap()
-        self.wait_for_element_not_present(*self._app_card_locator(app))
+        Wait(self.marionette).until(
+            expected.element_not_present(app_card))
```
Comment (davehunt) [inline tests/python/gaia-ui-tests/gaiatest/apps/system/regions/cards_view.py]: Hmm.. looks like this line is causing a failure in Gaia try. Perhaps `app_card` is not an element?

Comment (davehunt) [inline tests/python/gaia-ui-tests/gaiatest/apps/system/regions/cards_view.py]: Actually, the issue is that `element_not_present` performs a lookup and cannot take an existing element. Either reuse the locator or perhaps see if you can change this to displayed/not_displayed.

File: tests/python/gaia-ui-tests/gaiatest/apps/system/regions/utility_tray.py [hunk 33]
```diff
@@ -41,9 +40,9 @@ def tap_settings_button(self):
...
-        self.wait_for_element_not_displayed(*System(self.marionette)._utility_tray_locator)
+        self.accessibility.wheel(self.marionette.find_element(*self._grippy_locator), 'up')
+        Wait(self.marionette).until(
+            expected.element_not_displayed(*System(self.marionette)._utility_tray_locator))
```
Comment (davehunt) [inline tests/python/gaia-ui-tests/gaiatest/apps/system/regions/utility_tray.py]: `expected` is not imported.
